picard/tagger.py

```diff
@@ -287,9 +301,9 @@ def _run_init(self):
                     self.add_directory(decode_filename(file))
                 else:
                     files.append(decode_filename(file))
-            if files:
-                self.add_files(files)
             del self._cmdline_files
+            if files:
+                self.add_files([files])
```

Comment:
add_files now takes a list of lists of filenames. We group files in each directory into a list and then create a list of these - so we can queue these for guess_format and loading on a directory by directory basis so the user does not have to wait until we have guess_format on every file they have selected to load before they start appearing in the UI.